### PR TITLE
feat: add JSON Schema

### PR DESCRIPTION
**Pull request by jolheiser**
There are a few extraneous commits in here, let me know if you want them removed or in another PR.

1. Running https://github.com/mvdan/gofumpt changed a few things
2. I opted to use https://github.com/matryer/is as an assertion library. 

Both `mvdan/gofumpt` and `matryer/is` are personal preferences, so feel free to veto them.  
Of note, `mvdan/gofumpt` is fully compatible with `gofmt`, so it shouldn't cause conflicts.

-----

And of course, the point of the PR, please take a look at the schema to make sure it makes sense. I followed the spec as closely as I could.

image

### CODE REVIEW

**Review by spookyintheam — APPROVED** [review]

File: internal/utils/symlinks.go [hunk 1]
```diff
@@ -27,12 +27,9 @@ func makeLink(from string, to string, name string) string { // {{{
 	} else if PathExists(to + name) {
 		fmt.Println("Symlink already exists, removing and relinking...")
 		err := os.RemoveAll(to + name)
-
 		/* This remove a directory and replaces it with the new synlink in case it already exists.
 		The reason to use RemoveAll was that Remove cannot delete a directory if it is not empty.
-		On Unix, RemoveAll uses `rm -rf Dir/`*/
-
-		if err != nil {
+		On Unix, RemoveAll uses `rm -rf Dir/`*/ if err != nil {
```
Comment (nekowinston) [inline internal/utils/symlinks.go]: what happened here with the formatter? :monocle_face:

File: schema/schema.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
+{
+  "$schema": "https://json-schema.org/draft-07/schema",
+  "$id": "https://github.com/catppuccin/cli/raw/main/schema/schema.json",
+  "title": "Catppuccin config",
+  "description": "JSON schema for Catppuccin port configuration file",
+  "type": "object",
+  "additionalProperties": false,
```
Comment (nekowinston) [inline schema/schema.json]: I'd appreciate if the top schema checked for required values, but allowed additional properties.
IMO the CLI should only lint the `installer` section, would like @ghostx31 & @DakshG07 input on this.

**Review by nekowinston — CHANGES_REQUESTED** [review]
Thanks for the PR! Just need two minor clarifications / depending on those, one change.

File: internal/utils/symlinks.go (hunk 1 above)
Comment (jolheiser) [reply internal/utils/symlinks.go]: `gofumpt` seems to like doing this with multi-line comments.
I can change this back, shouldn't affect the code either way.

Comment (nekowinston) [reply internal/utils/symlinks.go]: just jumped out to me, seems like it'd be easier to miss.

File: schema/schema.json (hunk 2 above)
Comment (jolheiser) [reply schema/schema.json]:
```suggestion
  "additionalProperties": true,
```
I recall this conversation happening a while ago, so I will change this to `true` preemptively.  
wrt required fields, let me know if I should remove `app_name` and `binary_name` from the schema as well.

**Review by nekowinston — APPROVED** [review]

**Comment by spookyintheam** [comment]
Thanks for the PR!

**Pull request merged**